### PR TITLE
Update user active status and role

### PR DESCRIPTION
**Pull request by ShrawanBK**

### CODE REVIEW

File: src/components/UserRoles/index.tsx [hunk 1]
```diff
@@ -88,7 +125,45 @@ function UserRoles(props: UserRolesProps) {
...
+                    notify({ children: 'User active status updated successfully!' });
+                }
+            },
+        },
+    );
+
+    const loadingUsers = usersLoading || updateLoading;
+
+    const handleToggleUserActiveStatus = useCallback(
+        (id, newActiveStatus) => {
```
Comment (tnagorra) [inline src/components/UserRoles/index.tsx:152]: need typing here

File: src/components/UserRoles/index.tsx [hunk 2]
```diff
@@ -88,7 +125,45 @@ function UserRoles(props: UserRolesProps) {
...
+                variables: {
+                    id,
+                    isActive: newActiveStatus,
+                },
+            });
+        },
+        [toggleUserActiveStatus],
+    );
+
+    const handleShowUserRoleForm = useCallback((email) => {
```
Comment (tnagorra) [inline src/components/UserRoles/index.tsx:163]: Write typings

File: src/components/UserRoles/UserRoleForm.tsx [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+// eslint-disable-next-line @typescript-eslint/ban-types
+type WithId<T extends object> = T & {id: string };
+type FormType = PurgeNull<PartialForm<Omit<UpdateUserRoleMutationVariables, 'role'> & {role:BasicEntity['id']}>>;
+
+const schema: Schema<FormType> = {
+    fields: () => ({
+        id: [requiredCondition],
+        role: [requiredCondition],
+    }),
+};
```
Comment (tnagorra) [inline src/components/UserRoles/UserRoleForm.tsx:92]: We have a better way to write typings. We need to use that
Plus, id should have idCondition too

File: src/components/UserRoles/UserRoleForm.tsx [hunk 4]
```diff
@@ -0,0 +1,239 @@
...
+                }
+                const { results } = users;
+                if (!results) {
+                    return;
+                }
+                // NOTE: results is an array with only one object
+                onValueSet(removeNull({
+                    role: results[0].role,
+                    id: results[0].id,
+                }));
```
Comment (tnagorra) [inline src/components/UserRoles/UserRoleForm.tsx:142]: Need to check if there is atleast one result

File: src/components/UserRoles/UserRoleForm.tsx [hunk 5]
```diff
@@ -0,0 +1,240 @@
...
+                const { results } = users;
+                if (!results || results.length < 0) {
+                    return;
+                }
+
+                // NOTE: results is an array with only one object
+                onValueSet(removeNull({
+                    role: results[0].role,
+                    id: results[0].id,
+                }));
```
Comment (tnagorra) [inline src/components/UserRoles/UserRoleForm.tsx:143]:
```suggestion
                // NOTE: results is an array with only one object
                const firstRole = results?.[0];
                if (!firstRole) {
                    return;
                }


                onValueSet(removeNull({
                    role: firstRole.role,
                    id: firstRole.id,
                }));
```

File: src/components/UserRoles/UserActions.tsx [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+                <IoMdCreate />
+            </QuickActionButton>
+            {onToggleUserActiveStatus && (
+                <QuickActionConfirmButton
+                    name={undefined}
+                    onConfirm={handleToggleUserActiveStatus}
+                    title={activeStatus ? 'Deactivate' : 'Activate'}
+                    variant="danger"
+                    disabled={disabled || !onToggleUserActiveStatus}
+                    confirmationMessage="Change the user active status?"
```
Comment (tnagorra) [inline src/components/UserRoles/UserActions.tsx:76]:
```suggestion
                    confirmationMessage="Do you want to change the user active status?"
```

File: src/components/UserRoles/UserRoleForm.tsx [hunk 7]
```diff
@@ -0,0 +1,245 @@
...
+            enumValues {
+                name
+                description
+            }
+        }
+    }
+`;
+
+// TODO: find user role by id ( not email )
+// Update in backend required
```
Comment (tnagorra) [inline src/components/UserRoles/UserRoleForm.tsx:51]: I think we should do this before we merge this pull request

File: src/components/UserRoles/UserRoleForm.tsx [hunk 8]
```diff
@@ -0,0 +1,245 @@
...
+
+const schema: Schema<FormType> = {
+    fields: () => ({
+        id: [idCondition, requiredCondition],
+        role: [requiredCondition],
+    }),
+};
+
+interface UserRoleFormProps {
+    email: string | undefined;
```
Comment (tnagorra) [inline src/components/UserRoles/UserRoleForm.tsx:95]: Doesn't make sense to have non-required email and not have a create mutation.
Either this should be required or the form should handle "create" mutation

File: src/components/UserRoles/UserRoleForm.tsx (hunk 7 above)
Comment (ShrawanBK) [reply src/components/UserRoles/UserRoleForm.tsx:51]: @annshress

Comment (annshress) [reply src/components/UserRoles/UserRoleForm.tsx:51]: @ShrawanBK https://github.com/idmc-labs/helix-server/pull/61

**Pull request merged**